dist/dist-packages/linux/openziti-controller/postinstall.bash

```diff
@@ -0,0 +1,230 @@
+            echo "WARN: missing ZITI_CTRL_ADVERTISED_ADDRESS in ${ZITI_CTRL_BOOT_ENV_FILE}" >&2
+        fi
+    fi
+}
+
+promptPwd() {
+    # make ziti vars available in "ziti create config environment"
+    exportZitiVars
+    # shellcheck disable=SC1090 # compute the path to the identity file
+    source <(ZITI_HOME=/var/lib/ziti-controller ziti create config environment)
+    # do nothing if identity file has stuff in it
+    if [ -s "${ZITI_CTRL_DATABASE_FILE}" ]; then
+        echo "INFO: database exists in ${ZITI_CTRL_DATABASE_FILE}"
+    # prompt for password token if interactive, unless already answered
+    else
+        if ! [[ "${ZITI_BOOTSTRAP_DATABASE:-}" == true ]]; then
+            echo "INFO: ZITI_BOOTSTRAP_DATABASE is not true in ${ZITI_CTRL_SVC_FILE}" >&2
+        # do nothing if enrollment token is already defined in env file
+        elif [[ -n "${ZITI_PWD:-}" ]]; then
+            echo "INFO: ZITI_PWD is defined in ${ZITI_CTRL_BOOT_ENV_FILE} and will be used to init db during"\
+                    "next startup"
+        elif    grep -qE "^LoadCredential=ZITI_PWD:${ZITI_PWD_FILE}" "${ZITI_CTRL_SVC_FILE}" \
+                && [[ -s "${ZITI_PWD_FILE}" ]]; then
+            echo "INFO: ZITI_PWD is defined in ${ZITI_PWD_FILE} and will be used to"\
+                    "init db during next startup "
+        else
+            if ZITI_PWD="$(prompt "Enter the admin password: ")"; then
+                if [ -n "${ZITI_PWD:-}" ]; then
+                    echo "$ZITI_PWD" >| "${ZITI_PWD_FILE}"
+                fi
+            else
+                echo "WARN: missing ZITI_PWD; use LoadCredential in"\
+                        "${ZITI_CTRL_SVC_FILE} or set in ${ZITI_CTRL_BOOT_ENV_FILE}" >&2
+            fi
+        fi
+    fi
+}
+
+promptBootstrap() {
+    # if undefined, check previous answer in service unit or prompt for bootstrap, preserving default if no answer
+    if [[ -z "${ZITI_BOOTSTRAP:-}" ]]; then
+        if ZITI_BOOTSTRAP="$(prompt 'Generate a default config [y/N]: ' || echo 'false')"; then
+            if [[ "${ZITI_BOOTSTRAP}" =~ ^[yY]([eE][sS])?$ ]]; then
+                ZITI_BOOTSTRAP=true
+            elif [[ "${ZITI_BOOTSTRAP}" =~ ^[nN][oO]?$ ]]; then
+                ZITI_BOOTSTRAP=false
+            fi
+        fi
+        sed -Ei "s/^(ZITI_BOOTSTRAP)=.*/\1=${ZITI_BOOTSTRAP}/" "${ZITI_CTRL_SVC_ENV_FILE}"
+    fi
+    if [[ "${ZITI_BOOTSTRAP}" != true ]]; then
+        exit 0
+    fi
+}
+
+promptCtrlPort() {
+    # if undefined or default value in env file, prompt for router port, preserving default if no answer
+    if [[ -z "${ZITI_CTRL_ADVERTISED_PORT:-}" ]]; then
+        if ZITI_CTRL_ADVERTISED_PORT="$(prompt 'Enter the controller port [1280]: ' || echo '1280')"; then
+            sed -Ei "s/^(ZITI_CTRL_ADVERTISED_PORT)=.*/\1=${ZITI_CTRL_ADVERTISED_PORT}/" "${ZITI_CTRL_BOOT_ENV_FILE}"
+        fi
+    fi
+    if [[ "${ZITI_CTRL_ADVERTISED_PORT}" -le 1024 ]]; then
+        grantNetBindService
```

Comment:
> port numbers strictly less than 1024

You're correct!

That's good advice. I'm contemplating using overrides. The `systemctl edit` command that partially-automates overrides isn't scriptable, only interactive, but we can perform the same actions in a script, i.e., place the override.conf in `/etc` (permanent override) or `/run` (temp override), then do `daemon-reload`.

Still, your advice is to document instead of automate. I see what you mean about the limitations of using a scriptlet that's only triggered during install/upgrade. I'm also contemplating moving the login in that postinstall scriptlet to a standalone script to allow us to document conveniently re-running it at will, rather than ferreting it out of whatever corner it's stuffed by the package manager.

The story I've been telling myself is that the interactive user needs the most help, and this scriptlet will do the right thing for them during the install/upgrade. Still, it should use systemd unit overrides if it's going to modify the service's configuration, I agree. That's a simple enough enhancement that conforms better to systemd norms (never modifying the unit in `/lib`). The package scriptlet will do nothing for the more sophisticated users that install non-interactively or bring their own config instead of generating one, so we'll still need the doc for them.

Reply:
I implemented the override.conf in the last commit. It was a minor change. This has the benefit of separating the systemd service unit directives that are intended to be modified by the user (or the scriptlet) from those that are hard-coded for internal reasons.